apollo-federation/src/sources/connect/validation/mod.rs

```diff
@@ -581,6 +597,23 @@ impl Location {
             .map(|(line, column)| Location { line, column })?;
         Some(Range { start, end })
     }
+
+    fn from_abstract_type_node<T>(
+        node: &Node<T>,
+        sources: &SourceMap,
+        abstract_type: &str,
+    ) -> Option<Range<Self>> {
+        let node_location = node.location()?;
+        let source = sources.get(&node_location.file_id())?;
+        let start = source
+            .get_line_column(node_location.offset())
+            .map(|(line, column)| Location { line, column })?;
+        let end = source
+            .get_line_column(node_location.offset() + abstract_type.chars().count())
```

Comment:
This is clever! Seems like the "right" thing to do is to add something to `apollo-rs` that gives us the location of the keyword, but that'll take a while. This seems like a good workaround in the meantime.

The only risk, I think, is that `apollo-rs` changes the way they do locations, but then we should see our snapshots change/fail.